kura/org.eclipse.kura.driver.ble.sensortag.provider/src/main/java/org/eclipse/kura/internal/driver/ble/sensortag/TiSensorTag.java

```diff
@@ -0,0 +1,1503 @@
+                byte[] value = { ByteBuffer.allocate(4).putInt(status).array()[3] };
+                ioValueChar.writeValue(value);
+            } catch (KuraException e) {
+                logger.error("Switch off green led failed", e);
+            }
+        } else {
+            logger.info(IO_ERROR_MESSAGE);
+        }
+    }
+
+    /*
+     * Switch on buzzer
+     */
+    public void switchOnBuzzer() {
+        // Write "04" to switch on buzzer
+        if (this.cc2650) {
+            BluetoothLeGattCharacteristic ioValueChar;
+            try {
+                ioValueChar = this.gattServices.get(IO).findCharacteristic(TiSensorTagGatt.UUID_IO_SENSOR_VALUE);
+                int status = (int) ioValueChar.readValue()[0] | 0x04;
+                byte[] value = { ByteBuffer.allocate(4).putInt(status).array()[3] };
+                ioValueChar.writeValue(value);
+            } catch (KuraException e) {
+                logger.error("Switch on buzzer failed", e);
+            }
+        } else {
+            logger.info(IO_ERROR_MESSAGE);
+        }
+    }
+
+    /*
+     * Switch off buzzer
+     */
+    public void switchOffBuzzer() {
+        // Write "00" to switch off buzzer
+        if (this.cc2650) {
+            BluetoothLeGattCharacteristic ioValueChar;
+            try {
+                ioValueChar = this.gattServices.get(IO).findCharacteristic(TiSensorTagGatt.UUID_IO_SENSOR_VALUE);
+                int status = (int) ioValueChar.readValue()[0] & 0xFB;
+                byte[] value = { ByteBuffer.allocate(4).putInt(status).array()[3] };
+                ioValueChar.writeValue(value);
+            } catch (KuraException e) {
+                logger.error("Switch of buzzer failed", e);
+            }
+        } else {
+            logger.info(IO_ERROR_MESSAGE);
+        }
+    }
+
+    // ---------------------------------------------------------------------------------------------
+    //
+    // Auxiliary methods
+    //
+    // ---------------------------------------------------------------------------------------------
+
+    private String byteArrayToHexString(byte[] value) {
+        final char[] hexadecimals = "0123456789ABCDEF".toCharArray();
+        char[] hexValue = new char[value.length * 2];
+        for (int j = 0; j < value.length; j++) {
+            int v = value[j] & 0xFF;
+            hexValue[j * 2] = hexadecimals[v >>> 4];
+            hexValue[j * 2 + 1] = hexadecimals[v & 0x0F];
+        }
+        return new String(hexValue);
+    }
+
+    private int unsignedToSigned(int unsigned, int bitLength) {
+        int unsignedResult = 0;
+        if ((unsigned & 1 << bitLength - 1) != 0) {
+            unsignedResult = -1 * ((1 << bitLength - 1) - (unsigned & (1 << bitLength - 1) - 1));
+        }
+        return unsignedResult;
+    }
+
+    private static Integer shortSignedAtOffset(byte[] c, int offset) {
+        Integer lowerByte = c[offset] & 0xFF;
```

Comment:
Is there any specific reason of auto-boxing?

Reply:
No, I'll change to int.